app/routes/analysis/analysis.py

```diff
@@ -0,0 +1,63 @@
+"""Run analysis on registered datasets."""
+import json
+from typing import List, Optional
+from uuid import UUID
+
+from fastapi import APIRouter, Query, Path
+from fastapi.responses import ORJSONResponse
+
+from app.errors import InvalidResponseError
+from app.models.enum.analysis import RasterLayer
+from app.utils.geostore import get_geostore_geometry
+
+from ...models.enum.geostore import GeostoreOrigin
+from ...models.pydantic.responses import Response
+from app.settings.globals import RASTER_ANALYSIS_LAMBDA_NAME
+from app.utils.aws import get_lambda_client
+
+router = APIRouter()
+
+
+@router.get(
+    "/zonal/{geostore_id}",
+    response_class=ORJSONResponse,
+    response_model=Response,
+    tags=["Analysis"],
+)
+async def zonal_statistics(
+    *,
+    geostore_id: UUID = Path(..., title="Geostore ID"),
+    geostore_origin: GeostoreOrigin = Query(GeostoreOrigin.gfw,
+        title="Origin service of geostore ID"
+    ),
+    group_by: Optional[List[RasterLayer]] = Query([], title="Group By Layers"),
+    filters: Optional[List[RasterLayer]] = Query([], title="Filter Layers"),
+    sum: Optional[List[RasterLayer]] = Query([], title="Sum Layers"),
+    start_date: Optional[str] = Query(None, title="Start Date"),
+    end_date: Optional[str] = Query(None, title="End Date")
```

Comment:
Might it be possible/desirable to have the date format validated here? There's a regex in app/models/pydantic/metadata.py that could be of use.

Reply:
oh that's useful